src/main/java/com/study/cheeper/controller/CheepController.java

```diff
@@ -21,13 +23,16 @@
     @Autowired
     private CheepRepository cheepRepository;
 
+    @Autowired @Lazy
+    private User loggedUser;
+
     @PostMapping
     public String createNewCheep(NewCheepForm newCheepForm, RedirectAttributes attr) {
-        Cheep cheep = newCheepForm.toCheep(userRepository);
+        User author = userRepository.getOne(loggedUser.getId());
```

Comment:
Aqui... o lance é que o UserLogged que vem do Security não é managed... o getOne achei a melhor maneira porque não faz o select e volta o User managed...

Eu abri essa thread aqui: 
https://stackoverflow.com/questions/63161387/how-to-make-an-object-injected-by-spring-managed-in-a-jpa-context  só pra ver se alguém tinha alguma outra ideia hahaha mas no fundo os dois exemplos que eu coloquei na  pergunta são equivalentes... só muda que um é abstraído pelo spring data jpa... mas eu já tinha enviado pergunta então deixei hahah

Reply:
Que tal mover essa linha para dentro do producer para sempre receber o user managed?